recorder.c

```diff
@@ -799,6 +813,16 @@ static void recorder_format_entry(recorder_show_fn show,
     char *dst = buffer;
     char *dst_end = buffer + sizeof buffer;
 
+#define rsnprintf(...)                                                  \
+    snprintf(dst, dst_end >= dst ? dst_end - dst : 0,  __VA_ARGS__)
+
+#if RECORDER_64BIT
+#define REC_TS_FMT  "%.6f"
```

Comment:
you could have instead

#if RECODER_HZ == 1000000
#define REC_TS_FMT  "%.6f"
#elif RECODER_HZ == 1000
#define REC_TS_FMT  "%.3f"
#else 
#error RECORDER_HZ not supported
#endif

Reply:
Good idea. I went even a bit further after remembering that the code was designed to allow someone to pass a `RECORDER_HZ` at build time.